packages/core/src/blocks/ToggleWrapper/createToggleWrapper.ts

```diff
@@ -0,0 +1,191 @@
+import { ViewMutationRecord } from "@tiptap/pm/view";
+
+import { BlockNoteEditor } from "../../editor/BlockNoteEditor.js";
+import { Block } from "../defaultBlocks.js";
+
+export const createToggleWrapper = (
+  block: Block<any, any, any>,
+  editor: BlockNoteEditor<any, any, any>,
+  renderedElement: {
+    dom: HTMLElement;
+    contentDOM?: HTMLElement;
+    ignoreMutation?: (mutation: ViewMutationRecord) => boolean;
+    destroy?: () => void;
+  },
+  toggledState?: {
+    set: (block: Block<any, any, any>, isToggled: boolean) => void;
+    get: (block: Block<any, any, any>) => boolean;
+  },
+): {
+  dom: HTMLElement;
+  contentDOM?: HTMLElement;
+  ignoreMutation?: (mutation: ViewMutationRecord) => boolean;
+  destroy?: () => void;
+} => {
+  if ("isToggleable" in block.props && !block.props.isToggleable) {
+    return renderedElement;
+  }
+
+  if (!toggledState) {
+    toggledState = {
+      set: (block, isToggled: boolean) => {
+        window.localStorage.setItem(`toggle-${block.id}`, String(isToggled));
```

Comment:
Would prefer explicitness

```suggestion
        window.localStorage.setItem(`toggle-${block.id}`,isToggled ? "true" : "false");
```